pangaea/encoders/spectralgpt_encoder.py

```diff
@@ -198,21 +200,13 @@ def forward(self, image: dict[str, torch.Tensor]) -> list[torch.Tensor]:
             pos_embed = self.pos_embed[:, :, :]
         x = x + pos_embed
 
-        # reshape to [N, T, L, C] or [N, T*L, C]
-        requires_t_shape = (
-            len(self.blocks) > 0  # support empty decoder
-            and hasattr(self.blocks[0].attn, "requires_t_shape")
-            and self.blocks[0].attn.requires_t_shape
-        )
-        if requires_t_shape:
-            x = x.view([N, T, L, C])
-
         output = []
         for i, blk in enumerate(self.blocks):
             x = blk(x)
             if i in self.output_layers:
                 if self.cls_embed:
                     x = x[:, 1:]
+                x = x.view(N, T, L, C).transpose(2, 3).flatten(1, 2)
```

Comment:
SpectralGPT model doesn't work now, giving the error:
[rank0]:   File "/localhome/yjia/code/geofm-bench/pangaea/encoders/spectralgpt_encoder.py", line 212, in forward
[rank0]:     .view(
[rank0]: RuntimeError: view size is not compatible with input tensor's size and stride (at least one dimension spans across two contiguous subspaces). Use .reshape(...) instead.